### ISSUE TITLE
addRoutes problem when running 01-Basic example from new camel-k-example docs

### ISSUE THREAD

**Issue by michaelwjoyner** [apache/camel-k#1509]
I was working on the docs for 03-Saga didact and noticed some errors from Camel. I ran the 01-Basic example and the same errors were coming back.

Using minikube, namespace=basic, and most recent kamel binary.

```
Dev Mode -- Integration basic in phase "Running"
Dev Mode -- Condition "Ready" is "True" for Integration basic
Starting new integration via Kamel executable.
Deploying file Basic.java as integration basic
Dev Mode -- integration "basic" updated
Dev Mode -- Condition "IntegrationPlatformAvailable" is "True" for Integration basic: camel-k
Dev Mode -- Integration basic in phase "Initialization"
Dev Mode -- Condition "IntegrationKitAvailable" is "False" for Integration basic: creating a new integration kit
Dev Mode -- Integration basic in phase "Building Kit"
Dev Mode -- Condition "IntegrationKitAvailable" is "True" for Integration basic: kit-breokkdc3sp6imim0j80
Dev Mode -- Integration basic in phase "Deploying"
Dev Mode -- Condition "CronJobAvailable" is "False" for Integration basic: different controller strategy used (deployment)
Dev Mode -- Condition "DeploymentAvailable" is "True" for Integration basic: deployment name is basic
Dev Mode -- Condition "ServiceAvailable" is "False" for Integration basic: no http service required
Dev Mode -- Condition "ExposureAvailable" is "False" for Integration basic: no host or service defined
Dev Mode -- Integration basic in phase "Running"
Dev Mode -- Condition "Ready" is "True" for Integration basic
Dev Mode -- Progress: integration "basic" in phase Initialization
Dev Mode -- Condition "IntegrationPlatformAvailable" is "True" for Integration basic: camel-k
Dev Mode -- Progress: integration "basic" in phase Building Kit
Dev Mode -- Condition "IntegrationPlatformAvailable" is "True" for Integration basic: camel-k
Dev Mode -- Progress: integration "basic" in phase Deploying
Dev Mode -- Progress: integration "basic" in phase Running
Dev Mode -- [1] Monitoring pod basic-565cbffb86-thwh2[1] 2020-06-08 00:50:41.292 [32mINFO [m [main] ApplicationRuntime - Add listener: org.apache.camel.k.listener.ContextConfigurer@53de625d
[1] 2020-06-08 00:50:41.302 [32mINFO [m [main] ApplicationRuntime - Add listener: org.apache.camel.k.listener.RoutesConfigurer@dbd8e44
[1] 2020-06-08 00:50:41.303 [32mINFO [m [main] ApplicationRuntime - Add listener: org.apache.camel.k.listener.PropertiesConfigurer@3a0baae5
Dev Mode -- [1] 2020-06-08 00:50:41.322 [32mINFO [m [main] RuntimeSupport - Looking up loader for language: java
[1] 2020-06-08 00:50:41.324 [32mINFO [m [main] RuntimeSupport - Found loader org.apache.camel.k.loader.java.JavaSourceLoader@50468873 for language java from service definition
[1] Exception in thread "main" java.lang.NoSuchMethodError: 'void org.apache.camel.main.MainConfigurationProperties.addRoutesBuilder(org.apache.camel.RoutesBuilder)'
Dev Mode -- [1] at org.apache.camel.k.main.ApplicationRuntime.addRoutes(ApplicationRuntime.java:80)
[1] at java.base/java.util.Optional.ifPresent(Optional.java:183)
Dev Mode -- [1] at org.apache.camel.k.listener.RoutesConfigurer.load(RoutesConfigurer.java:99)
[1] at org.apache.camel.k.listener.RoutesConfigurer.load(RoutesConfigurer.java:63)
Dev Mode -- [1] at org.apache.camel.k.listener.RoutesConfigurer.accept(RoutesConfigurer.java:53)
Dev Mode -- [1] at org.apache.camel.k.listener.AbstractPhaseListener.accept(AbstractPhaseListener.java:32)
[1] at org.apache.camel.k.main.ApplicationRuntime$MainListenerAdapter.lambda$invokeListeners$0(ApplicationRuntime.java:176)
[1] at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
[1] at java.base/java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357)
Dev Mode -- [1] at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:485)
[1] at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
Dev Mode -- [1] at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
[1] at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
Dev Mode -- [1] at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
[1] at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
Dev Mode -- [1]
```

**Comment by astefanutti** [comment]
This should be fixed with #1485. You need to re-install the Camel catalogs, e.g.:

```
kubectl apply -f deploy/camel-catalog-1.3.0-quarkus.yaml
kubectl apply -f deploy/camel-catalog-1.3.0-main.yaml
```

**Comment by michaelwjoyner** [comment]
This fixed the problem above for those who run into the same problem.

kamel install --operator-image=camelk/camel-k:1.0.0 --maven-repository=https://repository.apache.org/content/repositories/orgapachecamel-1219 --olm=false

**Comment by michaelwjoyner** [comment]
> This should be fixed with #1485. You need to re-install the Camel catalogs, e.g.:
> 
> ```
> kubectl apply -f deploy/camel-catalog-1.3.0-quarkus.yaml
> kubectl apply -f deploy/camel-catalog-1.3.0-main.yaml
> ```

@astefanutti - Thank you, Nicola helped me and it's working
